crates/router/src/db/cache.rs

```diff
@@ -24,15 +24,24 @@ where
         .redis_conn()
         .map_err(Into::<errors::StorageError>::into)?;
     let redis_val = redis.get_and_deserialize_key::<T>(key, type_name).await;
+    let get_data_set_redis = || async {
+        let data = fun().await?;
+        redis
+            .serialize_and_set_key(key, &data)
+            .await
+            .change_context(errors::StorageError::KVError)?;
+        Ok::<_, error_stack::Report<errors::StorageError>>(data)
+    };
     match redis_val {
         Err(err) => match err.current_context() {
-            errors::RedisError::NotFound => {
-                let data = fun().await?;
+            errors::RedisError::NotFound => get_data_set_redis().await,
+            errors::RedisError::JsonDeserializationFailed => {
+                // In case of deserialization failure remove the old data and insert the new data
                 redis
-                    .serialize_and_set_key(key, &data)
+                    .delete_key(key)
```

Comment:
Instead of deleting and setting in 2 redis calls, is it possible to update the value in one redis call  ?